File: 프로그래머스/lv1/72410. 신규 아이디 추천/신규 아이디 추천.py
def solution(new_id):
    special_character = ["-", "_", "."]
    answer = ''
    new_id = new_id.lower()
    
    for c in new_id:
        if 48 <= ord(c) <= 57 or 97 <= ord(c) <= 122 or c in special_character:
            if answer:
                if answer[-1] == "." and c == ".":
                    continue
            answer += c
            
    if answer[0] == ".":
        answer = answer[1:]
    
    if len(answer) == 0:
        answer += "a"
    
    if answer[-1] == ".":
        answer = answer[:-1]
    
    if len(answer) > 15:
        answer = answer[:15]
        if answer[-1] == ".":
            answer = answer[:14]
    
    if len(answer) <= 2:
        while len(answer) != 3:
            answer += answer[-1]
            
    return answer

File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.js
function solution(new_id) {
    var answer = '';
    let temp = "";
    new_id = new_id.toLowerCase();
    const re = new RegExp(/[^a-z0-9-_.]/,"g")
    new_id = new_id.replace(re, "")
    const re1 = new RegExp(/[.]+/,"g")
    new_id = new_id.replace(re1, ".")
    
    if (new_id[0] === ".") new_id = new_id.slice(1)
    if (new_id.at(-1) === ".") new_id = new_id.slice(0, -1)
    if (new_id.length === 0) new_id = "a"
    if (new_id.length >= 16) new_id = new_id.slice(0, 15)
    if (new_id.at(-1) === ".") new_id = new_id.slice(0, -1)
    if (new_id.length <= 2) new_id += new_id.at(-1).repeat(3 - new_id.length)
    return new_id;
}